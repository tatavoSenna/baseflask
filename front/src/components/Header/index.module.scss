.siteLayout {
	background: #fff;
	display: flex;
	justify-content: space-between;
	padding: 0 24px;
}

.mobile {
	background: #001529 !important;
	padding-left: 50px;
}

.trigger {
  font-size: 18px;
  line-height: 64px;
  cursor: pointer;
	transition: color 0.1s;
	&:hover {
		color: #1890ff;
	}
}

.profile {
	cursor: pointer;
	transition: color 0.1s;
	background-color: #fff;
	display: flex;
	justify-content: space-between;
	align-items: center;
	&:hover {
		color: #1890ff;
	}
}

.profileMobile {
	color: #fff;
}

.logo {
	width: 100px;
}
