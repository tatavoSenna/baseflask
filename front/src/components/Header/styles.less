@import '../../styles/variables.less';

.header {
  position: absolute;
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: calc(100% - 80px);
  height: 80px;
  padding: 0 40px;
  left: 0;
  top: 0;

  &:focus {
    outline: 0;
  }

  div {
    display: none;

    p {
      font-size: 14px;
      font-weight: 500;

      &:last-child {
        font-size: 10px;
        text-transform: uppercase;
        margin-top: 5px;

        a {
          color: @color-blue-1;
          cursor: pointer;

          &:hover {
            opacity: 0.8;
          }
        }
      }
    }
  }

  svg {
    display: flex;
    cursor: pointer;

    &:hover {
      * {
        color: @color-blue-1;
      }
    }
  }
}

.header__logo {
  width: 140px;
}

.header__menu {
  display: flex;
  cursor: pointer;

  &--active {
    * {
      color: @color-blue-1;
    }
  }
}

@media (min-width: @screen-lg) {
  .header {
    width: calc(100% - 380px);

    div {
      display: flex;
      flex-direction: column;
    }

    svg {
      display: none;
    }
  }
}
