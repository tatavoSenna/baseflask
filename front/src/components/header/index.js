import React from 'react'
import { useDispatch } from 'react-redux'
import { useHistory } from 'react-router-dom'
import { Layout, Menu } from 'antd'

import { logout } from '~/states/modules/session'

function Head() {
	const history = useHistory()
	const dispatch = useDispatch()

	const handleLogout = () => {
		dispatch(logout({ history }))
	}

	const { Item } = Menu
	return (
		<Layout.Header className="site-layout-background" style={styles.header}>
			<Menu mode="horizontal" defaultSelectedKeys={['2']} align="end">
				<Item
					key="setting:1"
					onClick={() => {
						window.location.assign(
							process.env.REACT_APP_DOCUSIGN_OAUTH_URL +
								'/auth?response_type=code&scope=signature&client_id=' +
								process.env.REACT_APP_DOCUSIGN_INTEGRATION_KEY +
								'&redirect_uri=' +
								process.env.REACT_APP_DOCUSIGN_REDIRECT_URL
						)
					}}>
					Docusign connect
				</Item>
				<Item key="setting:2" onClick={handleLogout}>
					Sair
				</Item>
			</Menu>
		</Layout.Header>
	)
}

const styles = {
	header: {
		padding: 0,
	},
}

export default Head
