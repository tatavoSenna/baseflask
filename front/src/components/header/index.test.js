import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'

import Header from './index.js'

jest.mock('react-redux', () => ({
	useDispatch: () => jest.fn(),
}))

describe('<Header />', () => {
	it('Snapshot testing', () => {
		const { asFragment } = render(
			<Router>
				<Header />
			</Router>
		)

		expect(asFragment()).toMatchSnapshot()
	})
})
