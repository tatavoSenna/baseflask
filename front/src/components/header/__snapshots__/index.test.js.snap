// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Snapshot testing 1`] = `
<DocumentFragment>
  <header
    class="ant-layout-header site-layout-background"
    style="padding: 0px;"
  >
    <ul
      align="end"
      class="ant-menu ant-menu-light ant-menu-root ant-menu-horizontal"
      role="menu"
    >
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child"
        role="menuitem"
      >
        Docusign connect
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child"
        role="menuitem"
      >
        Sair
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-horizontal ant-menu-overflowed-submenu"
        role="menuitem"
        style="visibility: hidden; position: absolute; display: none;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="button"
        >
          <span>
            ···
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
  </header>
</DocumentFragment>
`;
